store/store.go

```diff
@@ -217,22 +214,35 @@ func (s *Store[H]) GetByHeight(ctx context.Context, height uint64) (H, error) {
 	if height == 0 {
 		return zero, errors.New("header/store: height must be bigger than zero")
 	}
+
+	if h, err := s.getByHeight(ctx, height); err == nil {
+		return h, nil
+	}
+
 	// if the requested 'height' was not yet published
 	// we subscribe to it
-	h, err := s.heightSub.Sub(ctx, height)
-	if !errors.Is(err, errElapsedHeight) {
-		return h, err
+	if head := s.contiguousHead.Load(); head == nil || height > (*head).Height() {
```

Comment:
heightSub does the same check inside Wait, so this one is redundant

Reply:
Oh, that's interesting. Done.